<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>knowledge_manager</class>
 <widget class="QWidget" name="knowledge_manager">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Knowledge Manager</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="title_label">
     <property name="text">
      <string>Knowledge Manager</string>
     </property>
     <property name="styleSheet">
      <string>font-size: 18px; font-weight: bold; padding: 10px;</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="button_layout">
     <item>
      <widget class="QPushButton" name="create_fact_button">
       <property name="text">
        <string>Create Fact</string>
       </property>
       <property name="icon">
        <iconset theme="list-add"/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="reload_facts_button">
       <property name="text">
        <string>Reload Facts</string>
       </property>
       <property name="icon">
        <iconset theme="view-refresh"/>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QComboBox" name="category_filter">
       <property name="placeholderText">
        <string>Filter by category...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="tag_filter">
       <property name="placeholderText">
        <string>Filter by tags...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="search_box">
       <property name="placeholderText">
        <string>Search facts...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="facts_table">
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <column>
      <property name="text">
       <string>Text</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Category</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Tags</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Confidence</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Verified</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Created</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Access Count</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Actions</string>
      </property>
     </column>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>create_fact_button</sender>
   <signal>clicked()</signal>
   <receiver>knowledge_manager</receiver>
   <slot>on_create_fact_clicked()</slot>
  </connection>
  <connection>
   <sender>reload_facts_button</sender>
   <signal>clicked()</signal>
   <receiver>knowledge_manager</receiver>
   <slot>on_reload_facts_clicked()</slot>
  </connection>
  <connection>
   <sender>search_box</sender>
   <signal>textChanged(QString)</signal>
   <receiver>knowledge_manager</receiver>
   <slot>on_search_changed(QString)</slot>
  </connection>
  <connection>
   <sender>category_filter</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>knowledge_manager</receiver>
   <slot>on_category_filter_changed(QString)</slot>
  </connection>
  <connection>
   <sender>tag_filter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>knowledge_manager</receiver>
   <slot>on_tag_filter_changed(QString)</slot>
  </connection>
 </connections>
 <slots>
  <slot>on_create_fact_clicked()</slot>
  <slot>on_reload_facts_clicked()</slot>
  <slot>on_search_changed(QString)</slot>
  <slot>on_category_filter_changed(QString)</slot>
  <slot>on_tag_filter_changed(QString)</slot>
 </slots>
</ui>
